src/Microsoft.AspNet.Http.Extensions/HeaderDictionaryTypeExtensions.cs

```diff
@@ -80,7 +80,55 @@ internal static void SetList<T>(this IHeaderDictionary headers, string name, ILi
             }
             else
             {
-                headers[name] = values.Select(value => value.ToString()).ToArray();
+
+                switch (values.Count)
+                {
+                    case 0:
```

Comment:
values.Count == 0 was already checked in the prior if block. Might as well make this `if..else if...else`.